### PR TITLE
Baseline Ruby project features

### PR DESCRIPTION
**Pull request by razor-x**
Incorporates boilerplate from https://github.com/makenew/rbgem (via merge). 

> For context, the makenew repos are open source projects I've maintained for a long time. They contain minimally opinionated boilerplate for bootstrapping new projects.

Stuff that comes for free with this:

- Proper rbenv support with .ruby-version
- GH Codespaces support
- EditorConfig
- Boilerplate gitignore
- Boilerplate for MIT License
- Boilerplate README sections
- More rake tasks
- Code coverage
- GitHub actions
  - Manual version release
  - Publishing new versions
  - Autoformat PRs
- The linter is now [standard](https://github.com/testdouble/standard).
  - This is zero config and seems to only differ from the existing linter rules around the spacing of curly braces.
  - As long as there are no strong opinions on this, switching to this linter removes config and delegates all opinions to standard.

Other included stuff:

- Semantic release support added. The CHANGELOG file was not included as release notes will be automatically generated by semantic release and added to the GH release notes. ([See how this looks over here](https://github.com/rxlabs/semanticgem/releases/tag/v2.2.0).)
- All changes made to `lib` are from the format action, there are no code changes: [`2ef190d` (#31)](https://github.com/seamapi/ruby/pull/31/commits/2ef190d5bcd5ed1e82766dc69ff43ef7976ec783)
- ~~The minimum Ruby version was increased to 3, were we intentionally supporting 2?~~ We are now testing against all major versions 2.6 and up.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -1,27 +1,43 @@
...
+  test:
     runs-on: ubuntu-latest
-    name: Ruby ${{ matrix.ruby }}
+    timeout-minutes: 30
     strategy:
       matrix:
         ruby:
-          - '3.1.1'
-
+          - '3.0'
```
Comment (seveibar) [inline .github/workflows/main.yml:18]:
```suggestion
          - '2.6.5'
          - '3.0'
```

**Comment by seveibar** [comment]
let's squash bc auditing but node modules in future we shouldn't and should have custom template repo

**Review by abimaelmartell — APPROVED** [review]
yeah, looks pretty good, just make sure to squash your commits

**Review by seveibar — APPROVED** [review]
let's squash, thanks for adding 2.6 test!

**Pull request merged**